polyA/_runners.py

```diff
@@ -0,0 +1,526 @@
+    # command line option to just output confidence values for
+    # single annotation instead of do whole algorithm
+    """
+    Calculate confidence values for single annotations instead of
+    running the entire algorithm.
+
+    :param alignments: list of alignments to run on
+    :param lambdaa: the value of lambda to use (from Easel)
+    """
+    subfams = []
+    subfam_rows = []
+    scores = []
+
+    for i, a in enumerate(alignments):
+        subfams.append(a.subfamily)
+        subfam_rows.append(i)
+        scores.append(a.score)
+
+    confidence_list = confidence_only(lambdaa, scores)
+    confidence_list, subfams_copy = zip(*sorted(zip(confidence_list, subfams)))
+
+    # TODO: Provide a writer to the function
+    stdout.write(f"query_label\tconfidence\n")
+    for i in range(len(subfams_copy) - 1, 0, -1):
+        if confidence_list[i]:
+            stdout.write(f"{subfams_copy[i]}\t{confidence_list[i]}\n")
+
+
+def _validate_target(target: Alignment) -> None:
+    # TODO: Switch to using a Logger
+    if target.chrom_length == 0:
+        Logger(__name__).warning(
+            """No chromosome position information found
+            (this is OK for artificial sequences but --viz and --heatmap will fail)"""
+        )
+    elif target.chrom_length <= 25:
+        raise RuntimeError("Target sequence length must be >25")
+    elif target.chrom_length < 1000:
+        Logger(__name__).warning(
+            """Target region is <1000 in length,
+            did you mean to use such a short region?"""
+        )
+
+
+def _handle_single_alignment(target: Alignment, print_matrix_pos: bool) -> None:
+    """
+    If there is only one subfam in the alignment file, no need
+    to run anything because we know that subfam is the annotation.
+    """
+    if print_matrix_pos:
+        stdout.write("start\tstop\tID\tname\n")
+        stdout.write("----------------------------------------\n")
+        stdout.write(
+            f"{0}\t{target.stop - target.start}\t1111\t{target.subfamily}\n"
+        )
+    else:
+        stdout.write("start\tstop\tID\tname\n")
+        stdout.write("----------------------------------------\n")
+        stdout.write(
+            f"{target.start}\t{target.stop}\t1111\t{target.subfamily}\n"
+        )
+    exit()
+
+
+def _change_probs(seq_count: int) -> Tuple[float, float, float]:
+    change_prob_log = log(CHANGE_PROB / (seq_count - 1))
+
+    # jumping in and then out of the skip state counts as 1 jump
+    change_prob_skip = change_prob_log / 2
+
+    # 5% of the jump penalty, staying in skip state for 20nt "counts" as one jump
+    same_prob_skip = change_prob_log / 30
+
+    return change_prob_log, change_prob_skip, same_prob_skip
+
+
+def run_full(
+    alignments: List[Alignment],
+    tandem_repeats: List[TandemRepeat],
+    chunk_size: int,
+    gap_ext: int,
+    gap_init: int,
+    lambdaa: float,
+    outfile_conf: Optional[TextIO],
+    outfile_viz: Optional[TextIO],
```

Comment:
This should use one file per shard

Reply:
@kmcarey is going to decide how to proceed on this, may want to make SODA able to handle a single file correctly